### PR TITLE
Fix ObjectID generation

### PR DESCRIPTION
**Pull request by jannekaisgg**
Use similar logic as the MongoDB java drivers, in other words use SecureRandom to generate the 5 bytes for the middle part of ObjectID.

### CODE REVIEW

File: bson/src/main/scala/types.scala [hunk 1]
```diff
@@ -492,15 +442,12 @@ object BSONObjectID {
...
-      // machine id, 3 first bytes of md5(macadress or hostname)
-      id(4) = machineId(0)
-      id(5) = machineId(1)
-      id(6) = machineId(2)
-
-      // 2 bytes of the pid or thread id. Thread id in our case. Low endian
-      val threadId = Thread.currentThread.getId.toInt
-      id(7) = (threadId & 0xFF).toByte
-      id(8) = (threadId >> 8 & 0xFF).toByte
+      id(4) = (randomVal1 >> 16 & 0xFF).toByte
```
Comment (cobbpg) [inline bson/src/main/scala/types.scala]: Technically you don't even need the `& 0xFF` part, since it happens automatically.

**Review by huljas — APPROVED** [review]

File: bson/src/main/scala/types.scala [hunk 2]
```diff
@@ -492,15 +442,12 @@ object BSONObjectID {
...
+      id(5) = (randomVal1 >> 8 & 0xFF).toByte
+      id(6) = (randomVal1 & 0xFF).toByte
```
Comment (MrBagheera) [inline bson/src/main/scala/types.scala]: Wouldn't it be more straightforward to have 5 byte variables initialized already in constructor? I also do not get `randomVal1` and `randomVal2` - wouldn't it be easier to throw random long and then pick 5 bytes from it?

File: bson/src/main/scala/types.scala [hunk 3]
```diff
@@ -379,63 +379,13 @@ class BSONObjectID private (private val raw: Array[Byte])
 
 object BSONObjectID {
   private val maxCounterValue = 16777216
-  private val increment = new java.util.concurrent.atomic.AtomicInteger(scala.util.Random.nextInt(maxCounterValue))
+  private val secureRandom = new java.security.SecureRandom()
+  private val increment = new java.util.concurrent.atomic.AtomicInteger(secureRandom.nextInt(maxCounterValue))
+  private val randomVal1 = secureRandom.nextInt(0x01000000)
```
Comment (cobbpg) [inline bson/src/main/scala/types.scala]: There's no need to put these here, they could be limited to the scope where they are used.

File: bson/src/main/scala/types.scala (hunk 2 above)
Comment (jannekaisgg) [reply bson/src/main/scala/types.scala]: That was directly from the Java drivers, but we don't need to expose the maching/process id part which they did. So just replaced it with nextBytes to return 5 random bytes.

**Review by MrBagheera — APPROVED** [review]
Please, create `0.12.x-sgg-fixes` branch off `0.12.x-synchroinize-history` and merge it there.

**Comment by jannekaisgg** [comment]
Merged from command line to 0.12.x-sgg-fixes

**Pull request closed without merging**